packages/core/src/loaders/imageLoader.ts

```diff
@@ -183,6 +219,170 @@ export function loadAndCacheImages(
   return allPromises;
 }
 
+/**
+ * Loads an image given an imageId and optional priority and returns a promise
+ * which will resolve to the loaded image object or fail if an error occurred.
+ * The image is stored in the cache.
+ *
+ * @param referencedImageId -  A Cornerstone Image Object's imageId
+ * @param options - Options to be passed to the Image Loader
+ *
+ * @returns Image Loader Object
+ */
+export function createAndCacheDerivedImage(
+  referencedImageId: string,
+  options: DerivedImageOptions = {},
+  preventCache = false
+): Promise<IImage> {
+  if (referencedImageId === undefined) {
+    throw new Error(
+      'createAndCacheDerivedImage: parameter imageId must not be undefined'
+    );
+  }
+
+  if (options.imageId === undefined) {
+    options.imageId = `derived:${uuidv4()}`;
```

Comment:
Can you switch to crypto.randomUUID() - that generates a v4 uuid without needing a library (might not have the exact name there)

Reply:
uuid4 is our implementation and is not reading a library, and uses crypto

```js
// prettier-ignore
// @ts-nocheck
/**
 * Generates a unique id that has limited chance of collision
 *
 * @see {@link https://stackoverflow.com/a/2117523/1867984|StackOverflow: Source}
 * @returns a v4 compliant GUID
 */
export default function uuidv4(): string {
  return ([1e7]+-1e3+-4e3+-8e3+-1e11).replace(/[018]/g, c =>
    (c ^ crypto.getRandomValues(new Uint8Array(1))[0] & 15 >> c / 4).toString(16)
  );
}

```